pyvista/core/pointset.py

```diff
@@ -415,19 +416,71 @@ def is_all_triangles(self):
 
         """
         # Need to make sure there are only face cells and no lines/verts
-        faces = self.faces  # grab once as this takes time to build
-        if not len(faces) or len(self.lines) > 0 or len(self.verts) > 0:
+        if not self.n_faces or self.n_lines or self.n_verts:
             return False
 
-        # All we have are faces, check if all faces are indeed triangles
-        if faces.size % 4 == 0:
-            return (faces[::4] == 3).all()
-        return False
+        # in VTK9, they use connectivity and offset rather than cell
+        # data.  Use the new API as this is faster
+        if _vtk.VTK9 and False:
+            # early return if not all triangular
+            if self._connectivity_array.size % 3:
+                return False
+
+            # next, check if there are three points per face
+            return np.all(np.diff(self._offset_array) == 3)
+
+        else:  # pragma: no cover
+            # All we have are faces, check if all faces are indeed triangles
+            faces = self.faces  # grab once as this takes time to build
+            if faces.size % 4 == 0:
+                return (faces[::4] == 3).all()
+            return False
 
     def __sub__(self, cutting_mesh):
         """Compute boolean difference of two meshes."""
         return self.boolean_difference(cutting_mesh)
 
+    @property
+    def _offset_array(self):
+        """Return the array used to store cell offsets."""
+        return _vtk.vtk_to_numpy(self.GetPolys().GetOffsetsArray())
+
+    @property
+    def _connectivity_array(self):
+        """Return the array with the point ids that define the cells connectivity."""
+        return _vtk.vtk_to_numpy(self.GetPolys().GetConnectivityArray())
```

Comment:
Should we raise ourselves for VTK < 9? I know this is a "private" method so every developer for themself, but perhaps that error would be clearer than an AttributeError (assuming `GetConnectivityArray()` was added in VTK 9).

Reply:
Raising a version error now.  Thanks for the suggestion.